frontend/src/components/custom-tools/document-manager/DocumentManager.jsx

```diff
@@ -209,7 +209,7 @@ function DocumentManager({ generateIndex, handleUpdateTool, handleDocChange }) {
   const updatePageAndDoc = (newPage) => {
     setPage(newPage);
     const newSelectedDoc = listOfDocs[newPage - 1];
-    handleDocChange(newSelectedDoc?.document_id);
+    handleDocChange(newSelectedDoc);
```

Comment:
```
if (newSelectedDoc) {
    handleDocChange(newSelectedDoc.document_id);
}
```
Can we do a check here to avoid undefined case

Reply:
Thanks for the catch, @athul-rs! Done.